WF-712: Updated repo for new toolbelt changes (#15)
spatialos/FlexibleProjectExample: docs/reference/server-worker-configuration.md

| `componentDelivery` | Optional | Specifies the delivery settings for the worker's components. Analogous to [`component_delivery` in the previous configuration version](https://docs.improbable.io/reference/latest/shared/worker-configuration/bridge-config#component-delivery). |
| `componentSettings` | Optional | Defines component specific settings. Analogous to [`component_settings` in the previous configuration version](https://docs.improbable.io/reference/latest/shared/worker-configuration/bridge-config#component-settings). |
| `permissions` | Optional | Specifies the worker's entity permissions. Analogous to [`permissions` in the previous configuration version](https://docs.improbable.io/reference/latest/shared/worker-configuration/permissions), with a difference that the permissions property here is a single permission object rather than an array. |

> | `launchConfiguration` | Required | Describes how to launch the worker both locally and in the cloud. It contains two fields: <br> - `localDeployment`, which specifies how to start the worker from a pre-built binary file on your local machine. See below for more information. <br> - `cloudDeployment`, which specifies how to start the worker from the uploaded project assembly. See below for more information. |

Reviewer: ```suggestion
| `launchConfiguration` | Required | Describes how to launch the worker instance both locally and in the cloud. It contains two fields: <br> - `localDeployment`, which specifies how to start the worker instance from a pre-built binary file on your local machine. See below for more information. <br> - `cloudDeployment`, which specifies how to start the worker instance from the uploaded project assembly. See below for more information. |
```

Author: Done.

#### Local deployment configuration

> The local deployment configuration specifies how to launch a server-worker locally. Below is an example excerpt:

Reviewer: ```suggestion
The local deployment configuration specifies how to launch a server-worker instance locally. Below is an example excerpt:
```

Author: Done.

```json
      "windows": {
        "localBuildDirectory": "bin/x64/ReleaseWindows",
        "command": "MyWorker.exe",
        "arguments": [
          "${IMPROBABLE_RECEPTIONIST_HOST}",
          "${IMPROBABLE_RECEPTIONIST_PORT}",
          "${IMPROBABLE_WORKER_ID}",
          "--my_additional_flag=true"
        ]
      },
      "macos": {
        "localBuildDirectory": "bin/x64/ReleaseMacOS",
        "command": "MyWorker.app",
        "arguments": [
          "${IMPROBABLE_RECEPTIONIST_HOST}",
          "${IMPROBABLE_RECEPTIONIST_PORT}",
          "${IMPROBABLE_WORKER_ID}",
          "--my_additional_flag=true"
        ]
      },
      "linux": {
        "localBuildDirectory": "bin/x64/ReleaseLinux",
        "command": "MyWorker",
        "arguments": [
          "${IMPROBABLE_RECEPTIONIST_HOST}",
          "${IMPROBABLE_RECEPTIONIST_PORT}",
          "${IMPROBABLE_WORKER_ID}",
          "--my_additional_flag=true"
        ]
      }
```

> There must be a configuration for each platform the worker will be run on. The platform name can be one of `windows`, `macos` or `linux`. Each of these can have the following fields:

Reviewer: ```suggestion
There must be a configuration for each platform the worker instance will be run on. The platform name can be one of `windows`, `macos` or `linux`. Each of these can have the following fields:
```

Author: I think this is fine as-is.

| `arguments` | Required | A list of arguments to provide to the base command. The arguments can contain placeholder strings, like `{IMPROBABLE_RECEPTIONIST_HOST}`, which are interpolated by SpatialOS when the worker is launched. For a full list of supported placeholder strings, see our documentation on [worker launch configurations](https://docs.improbable.io/reference/latest/shared/worker-configuration/launch-configuration). |

#### Cloud deployment configuration

> The cloud deployment configuration specifies how to launch a server-worker in the cloud. It's the same as the local deployment configuration at the moment, with the following exceptions:

Reviewer: ```suggestion
The cloud deployment configuration specifies how to launch a server-worker instance in the cloud. It's the same as the local deployment configuration at the moment, with the following exceptions:
```

Author: Done.

- A zip file containing the directory specified in `localBuildDirectory` will be uploaded to the cloud.
- We'll add additional fields in the future, allowing you to specify where the worker binaries are uploaded.

> Note: We only support the `linux` platform for cloud deployment configurations because workers are always run in an Linux environment in the cloud.

Reviewer: ```suggestion
Note: We only support the `linux` platform for cloud deployment configurations because worker instances are always run in an Linux environment in the cloud.
```

Author: Done.
